_posts/2021-04-09-springboot-autoconfigurations.md

```diff
@@ -0,0 +1,436 @@
+# How Spring Boot's AutoConfigurations Internally Work
+
+In this article, we will discuss and understand how Spring Boot's AutoConfigurations internally works
+
+## 1. Introduction
+Before we deep dive into Spring Boot's AutoConfiguration, **what really is Spring Boot?**
+
+*As per Spring Boot documentation:*
+> It takes an opinionated view of the Spring platform and third-party libraries so you can get started with minimum fuss.
+
+## 2. What is Spring Boot Auto Configuration?
+
+> Spring Boot auto-configuration attempts to automatically configure your Spring application based on the jar dependencies that we have added in ``pom.xml`` or ``build.gradle`` depends on the tool which we are using.
+
+In a nutshell, Spring Boot looks at libraries or frameworks available on available properties and libraries on the classpath. Based on these, it provides the basic configuration needed to configure the application with these libraries or frameworks. This is called ``Auto Configuration``.
```

Comment:
what does "available properties" mean in this context? Do you mean configuration properties? Please elaborate for the reader.